spec/dependabot/update_checkers/elm/elm_package_spec.rb

```diff
@@ -27,25 +27,81 @@
   let(:dependency) do
     Dependabot::Dependency.new(
       name: dependency_name,
-      version: current_version,
+      version: dependency_version,
       requirements: requirements,
       package_manager: "elm-package"
     )
   end
   let(:dependency_name) { "realWorld/ElmPackage" }
-  let(:current_version) { Dependabot::Utils::Elm::Version.new("2.2.0") }
+  let(:dependency_version) { Dependabot::Utils::Elm::Version.new("2.2.0") }
   let(:requirements) do
-    [{ file: "elm-package.json", requirement: "1.0.0 <= v <= 2.2.0",
-       groups: [], source: nil }]
+    [{
+      file: "elm-package.json",
+      requirement: string_req,
+      groups: [],
+      source: nil
+    }]
   end
+  let(:string_req) { "1.0.0 <= v <= 2.2.0" }
 
   let(:elm_package) do
     Dependabot::DependencyFile.new(
-      content: fixture("elm", "elm_package", "for_update_checking"),
+      content: fixture("elm", "elm_package", fixture_name),
       name: "Gemfile",
       directory: directory
     )
   end
+  let(:fixture_name) { "for_update_checking" }
+
+  describe "up_to_date?" do
+    subject { checker.up_to_date? }
+
+    context "with a requirement that is out of date, but needs a full unlock" do
+      let(:fixture_name) { "elm_css_and_datetimepicker_ranges" }
+      let(:dependency_name) { "NoRedInk/datetimepicker" }
+      let(:string_req) { "3.0.0 <= v <= 3.0.1" }
+      let(:dependency_version) { nil }
+      let(:elm_package_url) do
+        "http://package.elm-lang.org/packages/NoRedInk/datetimepicker/"
+      end
+      let(:elm_package_response) do
+        fixture("elm", "elm_package_responses", "NoRedInk-datetimepicker.html")
+      end
+
+      before do
+        stub_request(:get, elm_package_url).
+          to_return(status: 200, body: elm_package_response)
+      end
+
+      it { is_expected.to eq(false) }
+    end
+  end
+
+  describe "can_update?" do
+    subject { checker.can_update?(requirements_to_unlock: unlock_level) }
+    let(:unlock_level) { :own }
+
+    context "with a requirement that is out of date, but needs a full unlock" do
+      let(:fixture_name) { "elm_css_and_datetimepicker_ranges" }
+      let(:dependency_name) { "NoRedInk/datetimepicker" }
+      let(:string_req) { "3.0.0 <= v <= 3.0.1" }
+      let(:dependency_version) { nil }
+      let(:elm_package_url) do
+        "http://package.elm-lang.org/packages/NoRedInk/datetimepicker/"
+      end
+      let(:elm_package_response) do
+        fixture("elm", "elm_package_responses", "NoRedInk-datetimepicker.html")
+      end
+      let(:unlock_level) { :all }
+
+      before do
+        stub_request(:get, elm_package_url).
+          to_return(status: 200, body: elm_package_response)
+      end
+
+      it { is_expected.to eq(true) }
```

Comment:
This test is failing for me locally and I have no idea why. Once I've got it fixed I think this is ready for a final review and then good to go.

Reply:
taking a look

Reply:
pushed part of the fix, still doing something weird (returning a version instead of a bool)

Reply:
ok so the problem with :102 is we're falling into this if clause: https://github.com/NoRedInk/dependabot-core/blob/13c5446d8acd5c560048c2150232e26e9b690c0f/lib/dependabot/update_checkers/elm/elm_package.rb#L26-L27

I don't know why we check `dependency.appears_in_lockfile?` given Elm doesn't have lockfiles. Can we remove this?

Reply:
another weirdness is why the dependency reports as if it's appearing in a lockfile

Reply:
I may have fixed it, idk if appropriately!

Reply:
That's exactly the right fix! Realised it was probably something like that as soon as I stepped away from my computer, but had been driving me mental for a good hour!